app/controllers/rubygems_controller.rb

```diff
@@ -18,8 +18,13 @@ def index
   end
 
   def show
-    @latest_version = @rubygem.versions.most_recent
-    @versions       = @rubygem.public_versions(5)
+    if @rubygem
```

Comment:
This conditional is not obvious (without reading the rest of the code) that if `@rubygem` is nil, then the gem is blacklisted. Maybe a new before_action to check for blacklisted gems?